packages/headless/src/controllers/smart-snippet-questions-list/headless-smart-snippet-questions-list.ts

```diff
@@ -53,6 +76,34 @@ export interface SmartSnippetQuestionsList extends Controller {
    * @param identifier - The identifier of a document used to create the smart snippet.
    */
   collapse(identifier: QuestionAnswerDocumentIdentifier): void;
+  /**
+   * Selects the source, logging a UA event to the Coveo Platform if the source hadn't been selected before.
+   *
+   * In a DOM context, we recommend calling this method on all of the following events:
+   * * `contextmenu`
+   * * `click`
+   * * `mouseup`
+   * * `mousedown`
+   *
+   * @param identifier - The `questionAnswerId` of the smart snippet to collapse.
+   */
+  selectSource(identifier: string): void;
```

Comment:
What would be the alternative instead of receiving a string ? The full smart snippet question object ?

I think the ID is "fine", however I just want to know the possible alternative. The problem with the param is that it's vert generic. AKA: You have to read the doc to know which `identifier` were talking about.

Reply:
ah, unfortunately, I already started taking a string for the other function, so changing it would require deprecating the string in the other functions.

I do think taking the full related snippet object may have been better... At-least I don't think this is too bad either, since this function is on the same controller whose state returns the IDs, so I think it's pretty intuitive which ID belongs here.